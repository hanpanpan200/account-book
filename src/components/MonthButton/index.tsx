import React, { useMemo } from 'react';
import { getMonth, getYear } from 'utils/dateUtil';
import arrowDownIcon from 'assets/icons/arrow-down.svg';

import styles from './index.module.scss';

interface Props {
  date?: Date
  onClick: () => void
}

const MonthButton: React.FC<Props> = ({ date, onClick }) => {
  const monthLabel = useMemo(() => {
    const year = date ? getYear(date) : '-';
    const month = date ? getMonth(date) : '-';
    return `${year}年${month}月`;
  }, [date]);

  return (
    <div className={styles.container} onClick={onClick}>
      <span>{monthLabel}</span>
      <img src={arrowDownIcon} alt='arrowDownIcon' />
    </div>
  )
}

export default MonthButton;