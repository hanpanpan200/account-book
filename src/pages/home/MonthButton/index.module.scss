.container {
  display: flex;
  align-items: center;
  margin-top: 10px;

  span {
    font-size: 16px;
    color: white;
    margin-right: 5px;
    line-height: 30px;
  }

  img {
    width: 30px;
    height: 30px
  }
}