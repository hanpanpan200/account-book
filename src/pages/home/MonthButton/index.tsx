import React, { useMemo } from 'react';
import arrowDownIcon from 'assets/icons/arrow-down.svg';

import styles from './index.module.scss';
import { getMonth, getYear } from '../../../utils';

interface Props {
  date?: Date
  onClick: () => void
}

const MonthButton: React.FC<Props> = ({ date, onClick }) => {
  const monthLabel = useMemo(() => {
    const year = date ? getYear(date) : '-';
    const month = date ? getMonth(date) : '-';
    return `${year}年${month}月`;
  }, [date]);

  return (
    <div className={styles.container} onClick={onClick}>
      <span>{monthLabel}</span>
      <img src={arrowDownIcon} alt='arrowDownIcon' />
    </div>
  )
}

export default MonthButton;