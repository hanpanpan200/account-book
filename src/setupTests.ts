// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/react';

export const fireClickEvent = (element: Document | Element | Window | Node) => {
  fireEvent(element, new MouseEvent('click', {
    bubbles: true,
    cancelable: true
  }));
}