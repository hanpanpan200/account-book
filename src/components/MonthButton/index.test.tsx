import React from 'react';
import { render } from '@testing-library/react';
import { fireClickEvent } from 'setupTests';
import MonthButton from './index';

describe('MonthButton', () => {
  const date = new Date(2021, 6, 15);
  const mockedOnClick = jest.fn();

  it('should render correctly when date is provided', () => {
    const { container } = render(<MonthButton date={date} onClick={mockedOnClick} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when date is not provided', () => {
    const { container } = render(<MonthButton onClick={mockedOnClick} />);
    expect(container).toMatchSnapshot();
  });

  it('should trigger onClick when button is clicked', () => {
    const { container } = render(<MonthButton date={date} onClick={mockedOnClick} />);
    const divs = container.getElementsByTagName('div');
    expect(divs.length).toEqual(1);
    const divElement = divs.item(0) as Element;
    fireClickEvent(divElement);
    expect(mockedOnClick).toBeCalled();
  });
});


