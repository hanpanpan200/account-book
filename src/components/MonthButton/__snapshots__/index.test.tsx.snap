// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthButton should render correctly when date is not provided 1`] = `
<div>
  <div
    class="container"
  >
    <span>
      -年-月
    </span>
    <img
      alt="arrowDownIcon"
      src="arrow-down.svg"
    />
  </div>
</div>
`;

exports[`MonthButton should render correctly when date is provided 1`] = `
<div>
  <div
    class="container"
  >
    <span>
      2021年7月
    </span>
    <img
      alt="arrowDownIcon"
      src="arrow-down.svg"
    />
  </div>
</div>
`;
